internal_ws/ykshim/src/prod_api.rs

```diff
@@ -0,0 +1,90 @@
+use std::ffi::{c_void, CString};
+use std::os::raw::c_char;
+
+use ykcompile::CompiledTrace;
+use yktrace::sir::{SirTrace, SIR};
+use yktrace::tir::TirTrace;
+
+/// The different ways by which we can collect a trace.
+#[derive(Clone, Copy)]
+#[repr(u8)]
+#[allow(dead_code)]
+enum TracingKind {
+    /// Software tracing via ykrustc.
+    SoftwareTracing = 0,
+    /// Hardware tracing via ykrustc + hwtracer.
+    HardwareTracing = 1,
+}
+
+#[no_mangle]
+unsafe extern "C" fn __ykshim_start_tracing(tracing_kind: u8) -> *mut c_void {
+    let tracing_kind = match tracing_kind {
+        0 => yktrace::TracingKind::SoftwareTracing,
+        1 => yktrace::TracingKind::HardwareTracing,
+        _ => return std::ptr::null_mut(),
```

Comment:
Why not `unreachable!()`?

Reply:
Actually, I think returning an error to the client is preferable to crashing. We can deal with this in a later PR where we look at error handling across the whole API.